### PR TITLE
Make sure PowerShell DLLs used by Infra Client are properly installed

### PR DESCRIPTION
**Pull request by johnmccrae**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).
Signed-off-by: John McCrae <john.mccrae@progress.com>

### CODE REVIEW

File: omnibus/config/software/powershell-dlls.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+skip_transitive_dependency_licensing true
+license :project_license
+
+build do
+  block "Install windows powershell dlls" do
+    # Copy the chef gem's distro stuff over
+    chef_gem_path = File.expand_path("../..", shellout!("#{install_dir}/embedded/bin/gem which chef").stdout.chomp)
+
+    chef_module_dir = "#{install_dir}/embedded/bin"
```
Comment (tas50) [inline omnibus/config/software/powershell-dlls.rb:30]: This should be chef_dll_dir probably

Comment (johnmccrae) [reply omnibus/config/software/powershell-dlls.rb]: done

File: omnibus/config/software/chef-dlls.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    # Copy the chef gem's distro stuff over
+    chef_gem_path = File.expand_path("../..", shellout!("#{install_dir}/embedded/bin/gem which chef").stdout.chomp)
+    chef_module_dir = "#{install_dir}/embedded/bin"
+
+    require 'fileutils'
+    FileUtils.cp_r "#{chef_gem_path}/distro/ruby_bin_folder/AMD64/.", "#{chef_module_dir}", :verbose => true
+
+    # Dir.glob("#{chef_gem_path}/distro/ruby_bin_folder/AMD64/*").each do |file|
+    #   copy_file(file, chef_module_dir)
+    # end
```
Comment (lamont-granquist) [inline omnibus/config/software/chef-dlls.rb:36]: need to delete the commented out code

File: omnibus/config/software/chef-dlls.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+license :project_license
+
+build do
+  block "Install windows powershell dlls" do
+    # Copy the chef gem's distro stuff over
+    chef_gem_path = File.expand_path("../..", shellout!("#{install_dir}/embedded/bin/gem which chef").stdout.chomp)
+    chef_module_dir = "#{install_dir}/embedded/bin"
+
+    require 'fileutils'
+    FileUtils.cp_r "#{chef_gem_path}/distro/ruby_bin_folder/AMD64/.", "#{chef_module_dir}", :verbose => true
```
Comment (lamont-granquist) [inline omnibus/config/software/chef-dlls.rb:32]: should remove the string interpolation from `"#{chef_module_dir}"` and just use the bare variable.

File: omnibus/config/software/chef-dlls.rb (hunk 2 above)
Comment (johnmccrae) [reply omnibus/config/software/chef-dlls.rb:36]: doh...

File: omnibus/config/software/chef-dlls.rb (hunk 3 above)
Comment (johnmccrae) [reply omnibus/config/software/chef-dlls.rb]: done

**Review by marcparadise — APPROVED** [review]

**Pull request merged**